galaxy_importer/ansible_test/builders/local_image_build.py

```diff
@@ -0,0 +1,92 @@
+# (c) 2012-2020, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+import os
+import requests
+import subprocess
+import tempfile
+
+from galaxy_importer import config
+from galaxy_importer import exceptions
+from shutil import copy
+
+config_data = config.ConfigFile.load()
+cfg = config.Config(config_data=config_data)
+default_logger = logging.getLogger(__name__)
+
+
+class Build(object):
+    """Use docker/podman to build ansible-test image with artifact inside."""
+    def __init__(self, filepath, collection_name, logger=default_logger):
+        self.filepath = filepath
+        self.log = logger or default_logger
+        self.tag = f'galaxy-importer:{collection_name}'
+        self.working_dir = tempfile.TemporaryDirectory()
+
+    def build_image(self):
+        self.log.info('Building ContainerFile')
+        Build._build_dockerfile(self.working_dir.name)
+        Build._copy_collection_file(
+            dir=self.working_dir.name,
+            filepath=self.filepath
+        )
+
+        self.log.info('Building image...')
+        Build._build_image_with_artifact(
+            dir=self.working_dir.name,
+            tag=self.tag,
+        )
+        return self.tag
+
+    def cleanup(self):
+        self.log.info('Removing temporary files, image and container')
+        self.working_dir.cleanup()
+
+        cmd = ['docker', 'image', 'rm', '-f', self.tag]
+        try:
+            subprocess.run(cmd)
+        except subprocess.CalledProcessError as e:
+            raise exceptions.AnsibleTestError(
+                'An exception occurred in: {}, message={}'.format(' '.join(cmd), e.msg))
+
+    def _build_dockerfile(dir):
+        url = 'https://raw.githubusercontent.com/ansible/galaxy-importer/master/docker/ansible-test/Dockerfile'
+        file_location = os.path.join(dir, 'Dockerfile')
+        with requests.get(url, allow_redirects=True) as r:
+            with open(file_location, 'wb') as f:
+                f.write(r.content)
+        with open(file_location, 'r+') as f:
+            lines = f.readlines()
+            for index, line in enumerate(lines):
+                if 'ENV HOME' in line:
+                    lines.insert(index - 1, '\nCOPY archive.tar.gz /archive/archive.tar.gz\nRUN chown -R user1:root /archive\n')
```

Comment:
We can probably add `RUN chown -R user1:root /archive` to the base image used by OSD and local image execution, but this can be done later

Reply:
actually, could you put a TODO here as a reminder of ^?